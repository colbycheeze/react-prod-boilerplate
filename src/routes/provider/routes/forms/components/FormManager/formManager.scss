@import "~style/colors";

.wrapper {
  // 1px on top to fix spacing issue
  padding: 1rem 2rem 0;
  max-width: 1040px;
  margin: 0 auto;

  h1 {
    font-size: 2.25rem;
  }
}

.formsHeading {
  position: relative;
}

.allForms {
  position: absolute;
  top: 1rem;
  right: 0;
  font-size: 1.2rem;

  // TODO: optimize full forms manager to allow for mobile use as well
  // For now, hide toggle link to all forms on mobile
  @media screen and (max-width: 678px){
    display: none;
  }
}

.pageDescription {
  display: block;
  margin-bottom: 2rem;
}

.filter {
  margin: 1rem 0 2rem;
  width: 75%;
  min-width: 650px;
  display: flex;
  align-items: flex-end;

  :global .pt-label {
    flex: 1;
    margin-right: 2rem;
    margin-bottom: 0;
  }

  :global .pt-switch {
    margin: 0;
    margin-right: 2rem;
  }
}

// ------------------------------------
// Dialogs
// ------------------------------------
.disclaimer {
  color: #555;
  margin: 3rem 0 2rem;
}

.inputWrapper {
  width: 100%;

  span {
    font-style: italic;
  }

  input {
    height: 40px;
    line-height: 40px;
  }
}

.dialogButtonWrapper {
  display: flex;
  justify-content: center;
}

.prompt {
  margin-bottom: 2rem;
}

.nest {
  margin: 0 3rem;
}

.pendingForms {
  margin-top: 1rem;
  flex-flow: column nowrap;
  justify-content: space-between;
}

.pendingFormsHeader {
  display: flex;
  flex-direction: row;
  border-bottom: 2px solid $medspoke-dark;
  padding-bottom: 0.2rem;

  span {
    text-transform: uppercase;
    font-size: 0.8rem;
  }
}

.pendingFormRow {
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  border-bottom: 1px solid $medspoke-light;
  span {
    height: 4rem;
    display: flex;
    align-items: center;
  }

  &:hover {
    background: $medspoke-ultra-light;
  }
}

.nameCol {
  flex-grow: 5;
  flex-basis: 0;
  .icon {
    margin-right: 0.5rem;
    color: $medspoke-medium;
  }
}

.assignedCol {
  flex-grow: 1;
  flex-basis: 0;
  display: flex;
  justify-content: center;
}

.statusCol {
  flex-grow: 1;
  flex-basis: 0;
  display: flex;
  justify-content: center;
}

span.completionStatus {
  display: inline-block;
  height: 0.5rem;
  width: 0.5rem;
  border-radius: 100%;
  margin-right: 0.5rem;
  background: $intent-danger;
}

// Hide assigngedCol on mobile
@media screen and (max-width: 678px){
  span.assignedCol {
    display: none;
  }
  
  .statusCol{
    flex-grow: 2;
  }
  .pageSubtitle{
    font-size: 1.2rem;
  }
  span.nameCol{
    padding-right: 0.4rem;
  }
}

@media screen and (max-width: 338px){
  .pageSubtitle{
    font-size: 1.1rem;
  }
}
