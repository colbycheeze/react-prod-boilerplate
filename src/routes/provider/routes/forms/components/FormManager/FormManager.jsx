import { Link } from 'react-router'
import React, { Component, PropTypes as PT } from 'react'
import FlipMove from 'react-flip-move'
import { connect } from 'react-redux'
import { values } from 'lodash'
import fuzzy from 'fuzzy'
import { selectUserTypeFromRoute } from 'modules/location'

import { TextInput } from 'components/core/forms'
import { Switch } from '@blueprintjs/core'
import {
  toggleArchived,
  toggleShowAll,
} from 'modules/models/models'
import {
  selectForms,
  selectFormTemplates,
  selectFormsState,
  selectTemplatesState,
} from 'modules/models/models.selectors'
import { LoadingOverlay, Section } from 'components/core'

import FormTemplate from './FormTemplate'
import Form from './Form'
import classes from './formManager.scss'

const mapStateToProps = (state) => {
  const forms = values(selectForms(state))
  const formTemplates = values(selectFormTemplates(state))
  const hasLoaded = !!selectTemplatesState(state).sorted.length

  return {
    hasLoaded,
    formTemplates,
    forms,
    isAdmin: selectUserTypeFromRoute(state) === 'admin',
    showArchived: selectFormsState(state).showArchived,
    showAllTemplates: selectTemplatesState(state).showAll,
  }
}

const actions = {
  toggleArchived,
  toggleShowAll,
}

export class FormManager extends Component {
  static propTypes = {
    formTemplates: PT.arrayOf(PT.shape({
      id: PT.string.isRequired,
      isGlobal: PT.bool.isRequired,
    })).isRequired,

    forms: PT.arrayOf(PT.shape({
      id: PT.string.isRequired,
      title: PT.string.isRequired,
    })).isRequired,

    // from connect
    isAdmin: PT.bool.isRequired,
    hasLoaded: PT.bool.isRequired,
    showArchived: PT.bool.isRequired,
    showAllTemplates: PT.bool.isRequired,
    toggleArchived: PT.func.isRequired,
    toggleShowAll: PT.func.isRequired,
  }

  constructor(props) {
    super(props)
    this.state = {
      filterBy: '',
    }
  }

  componentDidMount() {
    this.filterInput.focus()
  }

  handleFilterChange = event => {
    this.setState({ filterBy: event.target.value })
  }

  renderHeading = () => {
    const { isAdmin } = this.props

    if (!isAdmin) {
      return (
        <div className={classes.formsHeading}>
          <h1>Fill Forms</h1>
          <span className={classes.allForms}>
            <Link to="/provider/forms">Pending Forms</Link>
          </span>
          <span className={classes.pageDescription}>
            Your forms section is where you can find all of the forms that you have been assigned to complete.
            You can also generate PDF copies to share with credentialing entities such as hospitals, medical facilities, and insurance payers.
          </span>
          <h3>All Forms</h3>
        </div>
      )
    }
  }

  renderDefaultList = () => {
    const { formTemplates } = this.props
    const globalTemplates = formTemplates.filter(template => template.isGlobal)
    const entityTemplates = formTemplates.filter(template => !template.isGlobal)
    const globalExists = !!globalTemplates.length
    const entityExists = !!entityTemplates.length

    return (
      <div>
        {globalExists &&
          <Section label="Global Forms">
            {globalTemplates.map(template => (
              <FormTemplate
                key={template.id}
                {...template}
              />
            ))}
          </Section>
        }

        {entityExists &&
          <Section label="Entity Forms">
            {entityTemplates.map(template => (
              <FormTemplate
                key={template.id}
                {...template}
              />
            ))}
          </Section>
        }
      </div>
    )
  }

  renderFilteredList = () => {
    const { filterBy } = this.state
    const { forms, formTemplates } = this.props
    const options = {
      pre: '<b>',
      post: '</b>',
      extract: result => result.title,
    }
    const formResults = fuzzy.filter(filterBy, forms, options)
    const templateResults = fuzzy.filter(filterBy, formTemplates, options)

    return (
      <div>
        <Section label="Filtered Form Templates">
          {templateResults.map(({ original, string }) => (
            <FormTemplate
              key={original.id}
              id={original.id}
              title={original.title}
              decoratedTitle={string}
            />
          ))}
        </Section>

        <Section label="Filtered Forms">
          <FlipMove
            duration={250}
            staggerDurationBy={20}
            staggerDelayBy={20}
            enterAnimation="accordianVertical"
            leaveAnimation="accordianVertical"
          >
            {formResults.map(({ original, string }) => (
              <Form
                key={original.id}
                id={original.id}
                decoratedTitle={string}
                isFilteredResult
              />
            ))}
          </FlipMove>
        </Section>
      </div>
    )
  }

  render = () => (
    <div className={classes.wrapper}>

      {this.renderHeading()}

      <div className={classes.filter}>
        <TextInput
          label="Filter Forms and Templates"
          inputRef={element => { this.filterInput = element }}
          leftIconName="filter"
          value={this.state.filterBy}
          onChange={this.handleFilterChange}
          placeholder="Filter forms and templates..."
          type="search"
        />

        <Switch
          label="Show archived"
          checked={this.props.showArchived}
          onChange={this.props.toggleArchived} // eslint-disable-line react/jsx-handler-names
        />

        <Switch
          label="Show all form templates"
          checked={this.props.showAllTemplates}
          onChange={this.props.toggleShowAll} // eslint-disable-line react/jsx-handler-names
        />
      </div>

      {!this.props.hasLoaded && <LoadingOverlay inline title="Loading Form Data..." />}

      {this.state.filterBy.length ?
        this.renderFilteredList()
        :
        this.renderDefaultList()
      }
    </div>
    )
}

export default connect(mapStateToProps, actions)(FormManager)
