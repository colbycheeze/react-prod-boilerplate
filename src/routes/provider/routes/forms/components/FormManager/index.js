import FormManager from './FormManager'

export default FormManager
