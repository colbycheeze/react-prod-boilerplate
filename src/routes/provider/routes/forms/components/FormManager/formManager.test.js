import React from 'react'
import { values } from 'lodash'
import { formsData, formTemplates } from 'modules/models/models.factory.js'
import { FormManager } from './FormManager'

const setup = () => {
  const spies = {
    toggleArchived: jest.fn(),
    toggleShowAll: jest.fn(),
  }

  const props = {
    formTemplates: values(formTemplates()),
    forms: values(formsData()).slice(0, -1), // don't pass in the form belonging to provider '2'
    showArchived: false,
    showAllTemplates: false,
    toggleArchived: spies.toggleArchived,
    toggleShowAll: spies.toggleShowAll,
    hasLoaded: true,
    isAdmin: false,
  }

  return { spies, props }
}

describe('<FormManager />', () => {
  it('Shows loader when data is fetching', () => {
    const { props } = setup()
    props.forms = []
    props.formTemplates = []
    props.hasLoaded = false
    const wrapper = shallow(<FormManager {...props} />)

    expect(wrapper).toMatchSnapshot()
  })

  it('renders non archived forms', () => {
    const { props } = setup()
    const wrapper = shallow(<FormManager {...props} />)

    expect(wrapper).toMatchSnapshot()
  })

  it('archived switch is flipped when showArchived is true', () => {
    const { props } = setup()
    props.showArchived = true

    const wrapper = shallow(<FormManager {...props} />)

    expect(wrapper).toMatchSnapshot()
  })

  it('renders filtered list', () => {
    const { props } = setup()

    const wrapper = shallow(<FormManager {...props} />)
    const filterInput = wrapper.find({ type: 'search' })
    filterInput.simulate('change', { target: { value: 'te' } })

    expect(wrapper).toMatchSnapshot()
  })

  it('calls toggleArchived when switch is clicked', () => {
    const { props, spies } = setup()
    const wrapper = shallow(<FormManager {...props} />)

    expect(spies.toggleArchived).not.toBeCalled()
    wrapper.find({ label: 'Show archived' }).simulate('change')
    expect(spies.toggleArchived).toBeCalled()
  })

  it('calls toggleShowAll when switch is clicked', () => {
    const { props, spies } = setup()
    const wrapper = shallow(<FormManager {...props} />)

    expect(spies.toggleShowAll).not.toBeCalled()
    wrapper.find({ label: 'Show all form templates' }).simulate('change')
    expect(spies.toggleShowAll).toBeCalled()
  })
})
