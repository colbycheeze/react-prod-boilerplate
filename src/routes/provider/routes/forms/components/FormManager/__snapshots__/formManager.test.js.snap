exports[`<FormManager /> Shows loader when data is fetching 1`] = `
<div
  className="wrapper">
  <div
    className="formsHeading">
    <h1>
      Fill Forms
    </h1>
    <span
      className="allForms">
      <Link
        onlyActiveOnIndex={false}
        style={Object {}}
        to="/provider/forms">
        Pending Forms
      </Link>
    </span>
    <span
      className="pageDescription">
      Your forms section is where you can find all of the forms that you have been assigned to complete. You can also generate PDF copies to share with credentialing entities such as hospitals, medical facilities, and insurance payers.
    </span>
    <h3>
      All Forms
    </h3>
  </div>
  <div
    className="filter">
    <TextInput
      className=""
      danger={false}
      inputRef={[Function]}
      isLabelHidden={false}
      label="Filter Forms and Templates"
      labelClass=""
      large={false}
      leftIconName="filter"
      onChange={[Function]}
      placeholder="Filter forms and templates..."
      primary={false}
      required={false}
      success={false}
      type="search"
      value=""
      warning={false} />
    <Blueprint.Switch
      checked={false}
      label="Show archived"
      onChange={[Function]} />
    <Blueprint.Switch
      checked={false}
      label="Show all form templates"
      onChange={[Function]} />
  </div>
  <LoadingOverlay
    backdropClassName="whiteOverlay"
    className=""
    inline={true}
    isOpen={true}
    title="Loading Form Data..." />
  <div />
</div>
`;

exports[`<FormManager /> archived switch is flipped when showArchived is true 1`] = `
<div
  className="wrapper">
  <div
    className="formsHeading">
    <h1>
      Fill Forms
    </h1>
    <span
      className="allForms">
      <Link
        onlyActiveOnIndex={false}
        style={Object {}}
        to="/provider/forms">
        Pending Forms
      </Link>
    </span>
    <span
      className="pageDescription">
      Your forms section is where you can find all of the forms that you have been assigned to complete. You can also generate PDF copies to share with credentialing entities such as hospitals, medical facilities, and insurance payers.
    </span>
    <h3>
      All Forms
    </h3>
  </div>
  <div
    className="filter">
    <TextInput
      className=""
      danger={false}
      inputRef={[Function]}
      isLabelHidden={false}
      label="Filter Forms and Templates"
      labelClass=""
      large={false}
      leftIconName="filter"
      onChange={[Function]}
      placeholder="Filter forms and templates..."
      primary={false}
      required={false}
      success={false}
      type="search"
      value=""
      warning={false} />
    <Blueprint.Switch
      checked={true}
      label="Show archived"
      onChange={[Function]} />
    <Blueprint.Switch
      checked={false}
      label="Show all form templates"
      onChange={[Function]} />
  </div>
  <div>
    <Section
      label="Global Forms">
      <Connect(FormTemplate)
        archived={false}
        createdAt="2016-12-28T13:30:13.547Z"
        formData={
          Object {
            "fields": Object {
              "2": Object {
                "editable": false,
                "form_field_type": "description_field",
                "hidden": false,
                "id": "2",
                "label": "Attestation acknowledgement",
                "mappings": Array [],
                "row": "1",
                "section": "999",
                "type": "description",
                "value": "Once you have reviewed your entries, please click or tap the “Submit” button. A digital copy of your form will automatically be generated and placed in your Files section!
          
          If you have any questions or issues, please reach out to the MedSpoke Customer Success team at credentialing@medspoke.com, or (210) 739-1417.",
              },
            },
            "mappings": Object {},
            "nextGroupId": 1,
            "nextId": 3,
            "rows": Object {
              "1": Object {
                "fields": Array [
                  "2",
                ],
                "id": "1",
                "section": "999",
              },
            },
            "sections": Object {
              "999": Object {
                "id": "999",
                "name": "Attestation",
                "rows": Array [
                  "1",
                ],
                "type": "attestation",
              },
            },
          }
        }
        id="1"
        isGlobal={true}
        title="Texas Standard Credentialing Application" />
      <Connect(FormTemplate)
        archived={false}
        createdAt="2016-12-28T13:30:13.547Z"
        formData={
          Object {
            "fields": Object {
              "2": Object {
                "editable": false,
                "form_field_type": "description_field",
                "hidden": false,
                "id": "2",
                "label": "Attestation acknowledgement",
                "mappings": Array [],
                "row": "1",
                "section": "999",
                "type": "description",
                "value": "Once you have reviewed your entries, please click or tap the “Submit” button. A digital copy of your form will automatically be generated and placed in your Files section!
          
          If you have any questions or issues, please reach out to the MedSpoke Customer Success team at credentialing@medspoke.com, or (210) 739-1417.",
              },
            },
            "mappings": Object {},
            "nextGroupId": 1,
            "nextId": 3,
            "rows": Object {
              "1": Object {
                "fields": Array [
                  "2",
                ],
                "id": "1",
                "section": "999",
              },
            },
            "sections": Object {
              "999": Object {
                "id": "999",
                "name": "Attestation",
                "rows": Array [
                  "1",
                ],
                "type": "attestation",
              },
            },
          }
        }
        id="2"
        isGlobal={true}
        title="California Credentialing Application" />
      <Connect(FormTemplate)
        archived={false}
        createdAt="2016-12-28T13:30:13.547Z"
        formData={
          Object {
            "fields": Object {
              "2": Object {
                "editable": false,
                "form_field_type": "description_field",
                "hidden": false,
                "id": "2",
                "label": "Attestation acknowledgement",
                "mappings": Array [],
                "row": "1",
                "section": "999",
                "type": "description",
                "value": "Once you have reviewed your entries, please click or tap the “Submit” button. A digital copy of your form will automatically be generated and placed in your Files section!
          
          If you have any questions or issues, please reach out to the MedSpoke Customer Success team at credentialing@medspoke.com, or (210) 739-1417.",
              },
            },
            "mappings": Object {},
            "nextGroupId": 1,
            "nextId": 3,
            "rows": Object {
              "1": Object {
                "fields": Array [
                  "2",
                ],
                "id": "1",
                "section": "999",
              },
            },
            "sections": Object {
              "999": Object {
                "id": "999",
                "name": "Attestation",
                "rows": Array [
                  "1",
                ],
                "type": "attestation",
              },
            },
          }
        }
        id="5"
        isGlobal={true}
        title="Template with no copies" />
      <Connect(FormTemplate)
        archived={true}
        createdAt="2016-12-28T13:30:13.547Z"
        formData={
          Object {
            "fields": Object {
              "2": Object {
                "editable": false,
                "form_field_type": "description_field",
                "hidden": false,
                "id": "2",
                "label": "Attestation acknowledgement",
                "mappings": Array [],
                "row": "1",
                "section": "999",
                "type": "description",
                "value": "Once you have reviewed your entries, please click or tap the “Submit” button. A digital copy of your form will automatically be generated and placed in your Files section!
          
          If you have any questions or issues, please reach out to the MedSpoke Customer Success team at credentialing@medspoke.com, or (210) 739-1417.",
              },
            },
            "mappings": Object {},
            "nextGroupId": 1,
            "nextId": 3,
            "rows": Object {
              "1": Object {
                "fields": Array [
                  "2",
                ],
                "id": "1",
                "section": "999",
              },
            },
            "sections": Object {
              "999": Object {
                "id": "999",
                "name": "Attestation",
                "rows": Array [
                  "1",
                ],
                "type": "attestation",
              },
            },
          }
        }
        id="6"
        isGlobal={true}
        title="Template with no copies" />
    </Section>
    <Section
      label="Entity Forms">
      <Connect(FormTemplate)
        archived={false}
        createdAt="2016-12-28T13:30:13.547Z"
        formData={
          Object {
            "fields": Object {
              "2": Object {
                "editable": false,
                "form_field_type": "description_field",
                "hidden": false,
                "id": "2",
                "label": "Attestation acknowledgement",
                "mappings": Array [],
                "row": "1",
                "section": "999",
                "type": "description",
                "value": "Once you have reviewed your entries, please click or tap the “Submit” button. A digital copy of your form will automatically be generated and placed in your Files section!
          
          If you have any questions or issues, please reach out to the MedSpoke Customer Success team at credentialing@medspoke.com, or (210) 739-1417.",
              },
            },
            "mappings": Object {},
            "nextGroupId": 1,
            "nextId": 3,
            "rows": Object {
              "1": Object {
                "fields": Array [
                  "2",
                ],
                "id": "1",
                "section": "999",
              },
            },
            "sections": Object {
              "999": Object {
                "id": "999",
                "name": "Attestation",
                "rows": Array [
                  "1",
                ],
                "type": "attestation",
              },
            },
          }
        }
        id="3"
        isGlobal={false}
        title="Oscar Payer Network Enrollment Application" />
      <Connect(FormTemplate)
        archived={false}
        createdAt="2016-12-28T13:30:13.547Z"
        formData={
          Object {
            "fields": Object {
              "2": Object {
                "editable": false,
                "form_field_type": "description_field",
                "hidden": false,
                "id": "2",
                "label": "Attestation acknowledgement",
                "mappings": Array [],
                "row": "1",
                "section": "999",
                "type": "description",
                "value": "Once you have reviewed your entries, please click or tap the “Submit” button. A digital copy of your form will automatically be generated and placed in your Files section!
          
          If you have any questions or issues, please reach out to the MedSpoke Customer Success team at credentialing@medspoke.com, or (210) 739-1417.",
              },
            },
            "mappings": Object {},
            "nextGroupId": 1,
            "nextId": 3,
            "rows": Object {
              "1": Object {
                "fields": Array [
                  "2",
                ],
                "id": "1",
                "section": "999",
              },
            },
            "sections": Object {
              "999": Object {
                "id": "999",
                "name": "Attestation",
                "rows": Array [
                  "1",
                ],
                "type": "attestation",
              },
            },
          }
        }
        id="4"
        isGlobal={false}
        title="MyER Texas Credentialing Application" />
    </Section>
  </div>
</div>
`;

exports[`<FormManager /> renders filtered list 1`] = `
<div
  className="wrapper">
  <div
    className="formsHeading">
    <h1>
      Fill Forms
    </h1>
    <span
      className="allForms">
      <Link
        onlyActiveOnIndex={false}
        style={Object {}}
        to="/provider/forms">
        Pending Forms
      </Link>
    </span>
    <span
      className="pageDescription">
      Your forms section is where you can find all of the forms that you have been assigned to complete. You can also generate PDF copies to share with credentialing entities such as hospitals, medical facilities, and insurance payers.
    </span>
    <h3>
      All Forms
    </h3>
  </div>
  <div
    className="filter">
    <TextInput
      className=""
      danger={false}
      inputRef={[Function]}
      isLabelHidden={false}
      label="Filter Forms and Templates"
      labelClass=""
      large={false}
      leftIconName="filter"
      onChange={[Function]}
      placeholder="Filter forms and templates..."
      primary={false}
      required={false}
      success={false}
      type="search"
      value="te"
      warning={false} />
    <Blueprint.Switch
      checked={false}
      label="Show archived"
      onChange={[Function]} />
    <Blueprint.Switch
      checked={false}
      label="Show all form templates"
      onChange={[Function]} />
  </div>
  <div>
    <Section
      label="Filtered Form Templates">
      <Connect(FormTemplate)
        decoratedTitle="<b>T</b><b>e</b>xas Standard Credentialing Application"
        id="1"
        title="Texas Standard Credentialing Application" />
      <Connect(FormTemplate)
        decoratedTitle="MyER <b>T</b><b>e</b>xas Credentialing Application"
        id="4"
        title="MyER Texas Credentialing Application" />
      <Connect(FormTemplate)
        decoratedTitle="<b>T</b><b>e</b>mplate with no copies"
        id="5"
        title="Template with no copies" />
      <Connect(FormTemplate)
        decoratedTitle="<b>T</b><b>e</b>mplate with no copies"
        id="6"
        title="Template with no copies" />
      <Connect(FormTemplate)
        decoratedTitle="Oscar Payer Ne<b>t</b>work <b>E</b>nrollment Application"
        id="3"
        title="Oscar Payer Network Enrollment Application" />
    </Section>
    <Section
      label="Filtered Forms">
      <FlipMovePropConverter
        delay={0}
        disableAllAnimations={false}
        duration={250}
        easing="ease-in-out"
        enterAnimation="accordianVertical"
        getPosition={[Function]}
        leaveAnimation="accordianVertical"
        maintainContainerHeight={false}
        staggerDelayBy={20}
        staggerDurationBy={20}
        typeName="div">
        <Connect(withRouter(Form))
          decoratedTitle="<b>T</b><b>e</b>st Test"
          id="2"
          isFilteredResult={true} />
        <Connect(withRouter(Form))
          decoratedTitle="<b>T</b><b>e</b>st Test"
          id="4"
          isFilteredResult={true} />
        <Connect(withRouter(Form))
          decoratedTitle="<b>T</b><b>e</b>st Test"
          id="6"
          isFilteredResult={true} />
        <Connect(withRouter(Form))
          decoratedTitle="<b>T</b><b>e</b>st Test"
          id="8"
          isFilteredResult={true} />
        <Connect(withRouter(Form))
          decoratedTitle="A Form <b>T</b>hat is Archiv<b>e</b>d"
          id="9"
          isFilteredResult={true} />
      </FlipMovePropConverter>
    </Section>
  </div>
</div>
`;

exports[`<FormManager /> renders non archived forms 1`] = `
<div
  className="wrapper">
  <div
    className="formsHeading">
    <h1>
      Fill Forms
    </h1>
    <span
      className="allForms">
      <Link
        onlyActiveOnIndex={false}
        style={Object {}}
        to="/provider/forms">
        Pending Forms
      </Link>
    </span>
    <span
      className="pageDescription">
      Your forms section is where you can find all of the forms that you have been assigned to complete. You can also generate PDF copies to share with credentialing entities such as hospitals, medical facilities, and insurance payers.
    </span>
    <h3>
      All Forms
    </h3>
  </div>
  <div
    className="filter">
    <TextInput
      className=""
      danger={false}
      inputRef={[Function]}
      isLabelHidden={false}
      label="Filter Forms and Templates"
      labelClass=""
      large={false}
      leftIconName="filter"
      onChange={[Function]}
      placeholder="Filter forms and templates..."
      primary={false}
      required={false}
      success={false}
      type="search"
      value=""
      warning={false} />
    <Blueprint.Switch
      checked={false}
      label="Show archived"
      onChange={[Function]} />
    <Blueprint.Switch
      checked={false}
      label="Show all form templates"
      onChange={[Function]} />
  </div>
  <div>
    <Section
      label="Global Forms">
      <Connect(FormTemplate)
        archived={false}
        createdAt="2016-12-28T13:30:13.547Z"
        formData={
          Object {
            "fields": Object {
              "2": Object {
                "editable": false,
                "form_field_type": "description_field",
                "hidden": false,
                "id": "2",
                "label": "Attestation acknowledgement",
                "mappings": Array [],
                "row": "1",
                "section": "999",
                "type": "description",
                "value": "Once you have reviewed your entries, please click or tap the “Submit” button. A digital copy of your form will automatically be generated and placed in your Files section!
          
          If you have any questions or issues, please reach out to the MedSpoke Customer Success team at credentialing@medspoke.com, or (210) 739-1417.",
              },
            },
            "mappings": Object {},
            "nextGroupId": 1,
            "nextId": 3,
            "rows": Object {
              "1": Object {
                "fields": Array [
                  "2",
                ],
                "id": "1",
                "section": "999",
              },
            },
            "sections": Object {
              "999": Object {
                "id": "999",
                "name": "Attestation",
                "rows": Array [
                  "1",
                ],
                "type": "attestation",
              },
            },
          }
        }
        id="1"
        isGlobal={true}
        title="Texas Standard Credentialing Application" />
      <Connect(FormTemplate)
        archived={false}
        createdAt="2016-12-28T13:30:13.547Z"
        formData={
          Object {
            "fields": Object {
              "2": Object {
                "editable": false,
                "form_field_type": "description_field",
                "hidden": false,
                "id": "2",
                "label": "Attestation acknowledgement",
                "mappings": Array [],
                "row": "1",
                "section": "999",
                "type": "description",
                "value": "Once you have reviewed your entries, please click or tap the “Submit” button. A digital copy of your form will automatically be generated and placed in your Files section!
          
          If you have any questions or issues, please reach out to the MedSpoke Customer Success team at credentialing@medspoke.com, or (210) 739-1417.",
              },
            },
            "mappings": Object {},
            "nextGroupId": 1,
            "nextId": 3,
            "rows": Object {
              "1": Object {
                "fields": Array [
                  "2",
                ],
                "id": "1",
                "section": "999",
              },
            },
            "sections": Object {
              "999": Object {
                "id": "999",
                "name": "Attestation",
                "rows": Array [
                  "1",
                ],
                "type": "attestation",
              },
            },
          }
        }
        id="2"
        isGlobal={true}
        title="California Credentialing Application" />
      <Connect(FormTemplate)
        archived={false}
        createdAt="2016-12-28T13:30:13.547Z"
        formData={
          Object {
            "fields": Object {
              "2": Object {
                "editable": false,
                "form_field_type": "description_field",
                "hidden": false,
                "id": "2",
                "label": "Attestation acknowledgement",
                "mappings": Array [],
                "row": "1",
                "section": "999",
                "type": "description",
                "value": "Once you have reviewed your entries, please click or tap the “Submit” button. A digital copy of your form will automatically be generated and placed in your Files section!
          
          If you have any questions or issues, please reach out to the MedSpoke Customer Success team at credentialing@medspoke.com, or (210) 739-1417.",
              },
            },
            "mappings": Object {},
            "nextGroupId": 1,
            "nextId": 3,
            "rows": Object {
              "1": Object {
                "fields": Array [
                  "2",
                ],
                "id": "1",
                "section": "999",
              },
            },
            "sections": Object {
              "999": Object {
                "id": "999",
                "name": "Attestation",
                "rows": Array [
                  "1",
                ],
                "type": "attestation",
              },
            },
          }
        }
        id="5"
        isGlobal={true}
        title="Template with no copies" />
      <Connect(FormTemplate)
        archived={true}
        createdAt="2016-12-28T13:30:13.547Z"
        formData={
          Object {
            "fields": Object {
              "2": Object {
                "editable": false,
                "form_field_type": "description_field",
                "hidden": false,
                "id": "2",
                "label": "Attestation acknowledgement",
                "mappings": Array [],
                "row": "1",
                "section": "999",
                "type": "description",
                "value": "Once you have reviewed your entries, please click or tap the “Submit” button. A digital copy of your form will automatically be generated and placed in your Files section!
          
          If you have any questions or issues, please reach out to the MedSpoke Customer Success team at credentialing@medspoke.com, or (210) 739-1417.",
              },
            },
            "mappings": Object {},
            "nextGroupId": 1,
            "nextId": 3,
            "rows": Object {
              "1": Object {
                "fields": Array [
                  "2",
                ],
                "id": "1",
                "section": "999",
              },
            },
            "sections": Object {
              "999": Object {
                "id": "999",
                "name": "Attestation",
                "rows": Array [
                  "1",
                ],
                "type": "attestation",
              },
            },
          }
        }
        id="6"
        isGlobal={true}
        title="Template with no copies" />
    </Section>
    <Section
      label="Entity Forms">
      <Connect(FormTemplate)
        archived={false}
        createdAt="2016-12-28T13:30:13.547Z"
        formData={
          Object {
            "fields": Object {
              "2": Object {
                "editable": false,
                "form_field_type": "description_field",
                "hidden": false,
                "id": "2",
                "label": "Attestation acknowledgement",
                "mappings": Array [],
                "row": "1",
                "section": "999",
                "type": "description",
                "value": "Once you have reviewed your entries, please click or tap the “Submit” button. A digital copy of your form will automatically be generated and placed in your Files section!
          
          If you have any questions or issues, please reach out to the MedSpoke Customer Success team at credentialing@medspoke.com, or (210) 739-1417.",
              },
            },
            "mappings": Object {},
            "nextGroupId": 1,
            "nextId": 3,
            "rows": Object {
              "1": Object {
                "fields": Array [
                  "2",
                ],
                "id": "1",
                "section": "999",
              },
            },
            "sections": Object {
              "999": Object {
                "id": "999",
                "name": "Attestation",
                "rows": Array [
                  "1",
                ],
                "type": "attestation",
              },
            },
          }
        }
        id="3"
        isGlobal={false}
        title="Oscar Payer Network Enrollment Application" />
      <Connect(FormTemplate)
        archived={false}
        createdAt="2016-12-28T13:30:13.547Z"
        formData={
          Object {
            "fields": Object {
              "2": Object {
                "editable": false,
                "form_field_type": "description_field",
                "hidden": false,
                "id": "2",
                "label": "Attestation acknowledgement",
                "mappings": Array [],
                "row": "1",
                "section": "999",
                "type": "description",
                "value": "Once you have reviewed your entries, please click or tap the “Submit” button. A digital copy of your form will automatically be generated and placed in your Files section!
          
          If you have any questions or issues, please reach out to the MedSpoke Customer Success team at credentialing@medspoke.com, or (210) 739-1417.",
              },
            },
            "mappings": Object {},
            "nextGroupId": 1,
            "nextId": 3,
            "rows": Object {
              "1": Object {
                "fields": Array [
                  "2",
                ],
                "id": "1",
                "section": "999",
              },
            },
            "sections": Object {
              "999": Object {
                "id": "999",
                "name": "Attestation",
                "rows": Array [
                  "1",
                ],
                "type": "attestation",
              },
            },
          }
        }
        id="4"
        isGlobal={false}
        title="MyER Texas Credentialing Application" />
    </Section>
  </div>
</div>
`;
